ibis/expr/rules.py

```diff
@@ -801,3 +807,120 @@ def _validate(self, args, i):
+                        'Arguments of subset schema must be instances of the '
+                        'Argument class (rules).')
+                if column_rule.name is None:
+                    raise ValueError(
+                        'Column rules must be named inside a table.')
+                try:
+                    # Use the argument's name to get
+                    column = arg[column_rule.name]
+                except IbisTypeError:
+                    if column_rule.optional:
+                        break
+                    else:
+                        raise IbisTypeError(
+                            'No column with name {}.'.format(column_rule.name))
+                try:
+                    # Arguments must validate the column
+                    column_rule.validate([column], 0)
+                except IbisTypeError as e:
+                    six.raise_from(
+                        IbisTypeError('Could not satisfy rule: {}.'.format(
+                            str(column_rule))), e)
+
+
+class Table(Argument):
+    """A table argument.
+
+    Parameters
+    ----------
+    name : str
+        The name of the table argument.
+    optional : bool
+        Whether this table argument is optional or not.
+    schema : TableColumnValidator
+        A validator for the table's columns. Only column subset validators are
+        currently supported. One can be created through the class method
+        ``Table.with_column_subset``. See the example for usage.
+    doc : str
+        A docstring to document this argument.
+    validator : Argument
+        Allows adding custom validation logic to this argument.
+
+    Examples
+    --------
+    The following op will accept an argument named ``'table'``. Note that the
+    ``schema`` argument specifies rules for columns that are required to be in
+    the table: ``time``, ``group`` and ``value1``. These must match the types
+    specified in the column rules. Column ``value2`` is optional, but if
+    present it must be of the specified type. The table may have extra columns
+    not specified in the schema.
+
+    >>> import ibis.expr.datatypes as dt
+    >>> import ibis.expr.rules as rules
+    >>> import ibis.expr.operations as ops
+    >>> class MyOp(ops.ValueOp):
+    ...    input_type = [
+    ...        rules.table(
+    ...            name='table',
+    ...            schema=rules.table.with_column_subset(
+    ...                rules.column(name='time', value_type=rules.number),
+    ...                rules.column(name='group', value_type=rules.number),
+    ...                rules.column(name='value1', value_type=rules.number),
+    ...                rules.column(name='value2', value_type=rules.number,
+    ...                             optional=True)))]
+    ...    output_type = rules.type_of_arg(0)
+    """
+    def __init__(self, name=None, optional=False, schema=None, doc=None,
+                 validator=None, allow_extra=False, **arg_kwds):
```

Comment:
Do we still need `allow_extra`? I thought we had decided to get rid of that.

Reply:
:( forgot to remove it here srry

Reply:
Removed it. :thumbsup: